docs/docs/hydroflow_plus/quickstart/first-dataflow.mdx

```diff
@@ -0,0 +1,45 @@
+---
+sidebar_position: 1
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import firstTenSrc from '!!raw-loader!../../../../template/hydroflow_plus/src/first_ten.rs';
+import firstTenExampleSrc from '!!raw-loader!../../../../template/hydroflow_plus/examples/first_ten.rs';
+import { getLines, extractOutput } from '../../../src/util';
+
+# Your First Dataflow
+Let's look a minimal example of a Hydroflow+ program. We'll start with a simple dataflow that prints out the first 10 natural numbers.
+
+:::tip
+
+We recommend using the Hydroflow+ template to get started with a new project:
+
+```bash
+#shell-command-next-line
+cargo install cargo-generate
+#shell-command-next-line
+cargo generate gh:hydro-project/hydroflow template/hydroflow_plus
+```
+
+:::
+
+## Writing a Dataflow
+
+Hydroflow+ programs are _explicit_ about where computation takes place. So our dataflow program takes a single `&Process` parameter which is a handle to the single machine our program will run on. We can use this handle to materialize a stream using `source_iter` (which emits values from a provided collection), and then print out the values using `for_each`.
+
+<CodeBlock language="rust" title="src/first_ten.rs">{getLines(firstTenSrc, 1, 7)}</CodeBlock>
```

Comment:
(Hmm I had a branch to add tagged lines instead of just line numbers, I wonder where that went)